src/Nordea/Components/StepIndicator.elm

```diff
@@ -0,0 +1,241 @@
+        , backgroundColor
+        , border3
+        , borderBox
+        , borderRadius
+        , bottom
+        , boxSizing
+        , center
+        , color
+        , column
+        , displayFlex
+        , flexDirection
+        , fontSize
+        , fontWeight
+        , height
+        , int
+        , lineHeight
+        , listStyleType
+        , margin
+        , none
+        , pct
+        , position
+        , relative
+        , rem
+        , solid
+        , textAlign
+        , top
+        , width
+        , zIndex
+        )
+import Html.Styled exposing (Html, div, li, p, span, styled, text, ul)
+import List.Extra as List
+import Nordea.Resources.Colors as Colors
+import Nordea.Resources.Fonts as Fonts
+import Nordea.Resources.Icons as Icons
+
+
+type StepIndicator
+    = StepIndicator Config
+
+
+type alias Config =
+    { stepNames : List String
+    , currentStep : Int
+    }
+
+
+init : List String -> Int -> StepIndicator
+init stepNames currentStep =
+    StepIndicator
+        { stepNames = stepNames
+        , currentStep = currentStep
+        }
+
+
+view : StepIndicator -> Html msg
+view (StepIndicator config) =
+    div []
+        [ viewSteps config.stepNames config.currentStep
+        ]
+
+
+enumerateList : List a -> List ( Int, a )
+enumerateList list =
+    List.zip (List.range 1 (List.length list)) list
+
+
+viewSteps : List String -> Int -> Html msg
+viewSteps stepNames currentStep =
+    let
+        list =
+            List.map (viewStepItem currentStep) (enumerateList stepNames)
+    in
+    styled ul listStyles [] list
+
+
+viewStepItem : Int -> ( Int, String ) -> Html msg
+viewStepItem currentStep ( item, stepName ) =
+    if item == 1 then
+        if item == currentStep then
+            styled li
+                listStylesLi
+                []
+                [ styled div divStylesCurrent [] [ text (String.fromInt item) ]
+                , styled p textStyles [] [ text stepName ]
+                ]
+
+        else
+            styled li
+                listStylesLi
+                []
+                [ styled div divStylesActive [] [ Icons.check ]
+                , styled p textStyles [] [ text stepName ]
+                ]
+
+    else if item < currentStep then
+        styled li
+            listStylesLi
+            []
+            [ styled div divStylesActive [] [ Icons.check ]
+            , styled p textStyles [] [ text stepName ]
+            , viewLineActive
+            ]
+
+    else if item == currentStep then
+        styled li
+            listStylesLi
+            []
+            [ styled div divStylesCurrent [] [ text (String.fromInt item) ]
+            , styled p textStyles [] [ text stepName ]
+            , viewLineActive
+            ]
+
+    else
+        styled li
+            listStylesLi
+            []
+            [ styled div divStyles [] [ text (String.fromInt item) ]
+            , styled p textStyles [] [ text stepName ]
+            , viewLine
+            ]
```

Comment:
The first `if` case could be eliminated by conditionally showing the line element
```elm
showIf : Bool -> Html msg -> Html msg
showIf pred elem =
    if pred then
        elem

    else
        text ""
```

```suggestion
viewStepItem : Int -> ( Int, String ) -> Html msg
viewStepItem currentStep ( item, stepName ) =
    if item < currentStep then
        styled li
            listStylesLi
            []
            [ styled div divStylesActive [] [ Icons.check ]
            , styled p textStyles [] [ text stepName ]
            , showIf (item /= 1) viewLineActive
            ]

    else if item == currentStep then
        styled li
            listStylesLi
            []
            [ styled div divStylesCurrent [] [ text (String.fromInt item) ]
            , styled p textStyles [] [ text stepName ]
            , showIf (item /= 1) viewLineActive
            ]

    else
        styled li
            listStylesLi
            []
            [ styled div divStyles [] [ text (String.fromInt item) ]
            , styled p textStyles [] [ text stepName ]
            , showIf (item /= 1) viewLine
            ]
```